charts/dapr/README.md

```diff
@@ -172,22 +172,23 @@ The Helm chart has the follow configuration options that can be supplied:
-| `dapr_scheduler.securityContext.runAsNonRoot`  | Boolean value for `securityContext.runAsNonRoot`. You may have to set this to `false` when running in Minikube | `true`                                      |
-| `dapr_scheduler.securityContext.fsGroup`       | Integer value for `securityContext.fsGroup`. Useful for adding the Scheduler process to the file system group that can write to the mounted database volume.              | `65532`                                      |
-| `dapr_scheduler.resources`                     | Value of `resources` attribute. Can be used to set memory/cpu resources/limits. See the section "Resource configuration" above. Defaults to empty                         | `{}`                                         |
-| `dapr_scheduler.debug.enabled`                 | Boolean value for enabling debug mode                                                                                                                                     | `{}`                                         |
-| `dapr_scheduler.statefulsetAnnotations`        | Custom annotations for Dapr Scheduler Statefulset                                                                                                                         | `{}`                                         |
-| `dapr_scheduler.service.annotations`           | Custom annotations for "dapr-scheduler-server" Service resource                                                                                                           | `{}`                                         |
-| `dapr_scheduler.extraEnvVars`                  | Dictionary (key: value pairs) to use as extra environment variables in the injected sidecar containers (e.g. `my-env-var: "my-val"`, etc)                                 | `{}`                                         |
+| Parameter                                     | Description                                                                                                                                                  | Default                                      |
+|-----------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------|
+| `dapr_scheduler.replicaCount`                 | Number of replicas                                                                                                                                           | `1`                                          |
+| `dapr_scheduler.ha.replicaCount`              | Number of replicas when in HA mode (should be odd number or helm will fail)                                                                                  | `3`                                          |
```

Comment:
IMHO this logic is unnecessarily complex. even `replicaCount` must be an odd number. And even among odd numbers, it can either be 1, 3, or 5 (more than 5 doesn't make sense).

You could just keep `dapr_scheduler.replicaCount`, with a default of `null`. If HA mode is enabled, then NULL = 3, otherwise NULL = 1. If set manually, must be 1, 3, or 5

Reply:
I agree, this logic makes sense to me re: single count value field, defaulting on `null` and constrained on >=5 odd int.

We should also have this written up in the docs under some “productionize” section. Along with a ref to the wip persistent volume docs https://github.com/dapr/docs/pull/4256

Reply:
I agree this logic is too complex. Let's just keep the "ha" flag, like in placement, and revisit this in 1.15. We can remove the need for odd number of replicas if we configure scheduler to run "stateless" (without Etcd) for anything > 3. It means that we will allow repliacCount to be anything >= 3 or HA mode and anything >= 1 for non-HA mode.